### PR TITLE
Fix MSVC compilation errors

### PR DESCRIPTION
**Pull request by kosz78**
This changes are resolved issue #11373 and error after 3528b1e571dee24917b0141232f135e29bf088ba commit

### CODE REVIEW

File: core/os/file_access.h [hunk 1]
```diff
@@ -140,7 +140,7 @@ class FileAccess {
 
 	virtual Error reopen(const String &p_path, int p_mode_flags); ///< does not change the AccessType
 
-	virtual Error _chmod(const String &p_path, int p_mod) {}
```
Comment (akien-mga) [inline core/os/file_access.h]: Should it return `OK` or `FAILED` by default? @marcelofg55

File: modules/gdnative/gdnative.cpp [hunk 2]
```diff
@@ -40,12 +40,17 @@
 const String init_symbol = "godot_gdnative_init";
 const String terminate_symbol = "godot_gdnative_terminate";
```
Comment (akien-mga) [inline modules/gdnative/gdnative.cpp]: CC @karroffel @touilleMan for review.

File: core/os/file_access.h (hunk 1 above)
Comment (marcelofg55) [reply core/os/file_access.h]: Should return FAILED by default. I missed that.

**Comment by touilleMan** [comment]
@akien-mga Given GODOT_GDNATIVE_API_FUNCTIONS is ordered, can't we just replace `api_struct` definition by:
```cpp
#define GDAPI_FUNC(name, ret_type, ...) name,
#define GDAPI_FUNC_VOID(name, ...) name,

const godot_gdnative_api_struct api_struct = {
	GODOT_GDNATIVE_API_FUNCTIONS
};
```

**Comment by kosz78** [comment]
I have updated code according comments

**Comment by BastiaanOlij** [comment]
This seems to work for me, not sure how to test it beyond compiling.. :)

**Review by karroffel — APPROVED** [review]

**Pull request merged**